packages/eas-cli/src/credentials/ios/actions/AscApiKeyUtils.ts

```diff
@@ -1,15 +1,123 @@
 import chalk from 'chalk';
 import fs from 'fs-extra';
+import { nanoid } from 'nanoid';
 import path from 'path';
-import * as uuid from 'uuid';
 
 import Log, { learnMore } from '../../../log';
-import { promptAsync } from '../../../prompts';
-import { getCredentialsFromUserAsync } from '../../utils/promptForCredentials';
+import { confirmAsync, promptAsync } from '../../../prompts';
+import { CredentialsContext } from '../../context';
+import {
+  getCredentialsFromUserAsync,
+  shouldAutoGenerateCredentialsAsync,
+} from '../../utils/promptForCredentials';
 import { AscApiKey } from '../appstore/Credentials.types';
-import { AscApiKeyPath, MinimalAscApiKey, ascApiKeyMetadataSchema } from '../credentials';
+import {
+  AscApiKeyPath,
+  MinimalAscApiKey,
+  ascApiKeyIdSchema,
+  ascApiKeyIssuerIdSchema,
+} from '../credentials';
+import { isAscApiKeyValidAndTrackedAsync } from '../validators/validateAscApiKey';
 
-export async function promptForAscApiKeyAsync(): Promise<AscApiKeyPath> {
+export enum AppStoreApiKeyPurpose {
+  SUBMISSIONS_SERVICE = 'Submissions',
+  UNKNOWN = 'Unknown',
+}
+
+export async function promptForAscApiKeyPathAsync(ctx: CredentialsContext): Promise<AscApiKeyPath> {
+  const { keyId, keyP8Path } = await promptForKeyP8AndIdAsync();
+
+  const bestEffortIssuerId = await getBestEffortIssuerIdAsync(ctx, keyId);
+  if (bestEffortIssuerId) {
+    Log.log(`Detected Issuer ID: ${bestEffortIssuerId}`);
+    return { keyId, issuerId: bestEffortIssuerId, keyP8Path };
+  }
+  const issuerId = await promptForIssuerIdAsync();
+  return { keyId, issuerId, keyP8Path };
+}
+
+export async function promptForIssuerIdAsync(): Promise<string> {
+  Log.log(chalk.bold('An App Store Connect Issuer ID is required'));
+  Log.log(
+    `If you're not sure what this is or how to find yours, ${learnMore(
+      'https://expo.fyi/asc-issuer-id'
+    )}`
+  );
+
+  // Do not perform uuid validation - Apple's issuerIds are not RFC4122 compliant
+  const { issuerId } = await getCredentialsFromUserAsync(ascApiKeyIssuerIdSchema, {});
+  return issuerId;
+}
+
+export async function getMinimalAscApiKeyAsync(ascApiKey: AscApiKey): Promise<MinimalAscApiKey> {
+  return {
+    ...ascApiKey,
+    issuerId: ascApiKey.issuerId ?? (await promptForIssuerIdAsync()),
+  };
+}
+
+export async function provideOrGenerateAscApiKeyAsync(
+  ctx: CredentialsContext,
+  purpose: AppStoreApiKeyPurpose
+): Promise<MinimalAscApiKey> {
+  if (ctx.nonInteractive) {
+    return await generateAscApiKeyAsync(ctx, purpose);
+  }
+
+  const userProvided = await promptForAscApiKeyAsync(ctx);
+  if (!userProvided) {
+    return await generateAscApiKeyAsync(ctx, purpose);
+  }
+
+  if (!ctx.appStore.authCtx) {
+    Log.warn('Unable to validate App Store Connect Api Key due to insufficient Apple Credentials');
```

Comment:
IMO this sounds weird. Instead of "Apple Credentials" (I think this notion is not defined anywhere) I'd say something like "you chose not to authenticate with Apple".